web/bfex/components/data_pipeline/tasks/researchid.py

```diff
@@ -34,60 +28,54 @@ def run(self, data):
         :return: last faculty member handled
         """
         
-        no_text_count = 0
-        for faculty in data:
+        faculty = data
+        if isinstance(faculty, str):
+            faculty_name = faculty
+        else:
             faculty_name = faculty.name
-    
-            search_results = Faculty.search().query('match', name=faculty_name).execute()
-            if len(search_results) > 1:
-                # Shouldn't happen, but could.
-                raise WorkflowException("Professor id is ambiguous during search ... More than 1 result")
 
-            search_dup = Document.search().query('match', faculty_id=faculty.faculty_id).query("match", source="ResearchId")
-            search_dup.delete()
-            faculty = search_results[0]
-            if faculty.research_id is not None:
-                
-                scraper = ScraperFactory.create_scraper(faculty.research_id, ScraperType.RESEARCHID)
-                scrapps = scraper.get_scrapps()
+        search_results = Faculty.search().query('match', name=faculty_name).execute()
+        if len(search_results) > 1:
+            # Shouldn't happen, but could.
+            raise WorkflowException("Professor id is ambiguous during search ... More than 1 result")
+
+        search_dup = Document.search().query('match', faculty_id=faculty.faculty_id).query("match", source="ResearchId")
+        search_dup.delete()
+        faculty = search_results[0]
+        if faculty.research_id is not None:
+            
+            scraper = ScraperFactory.create_scraper(faculty.research_id, ScraperType.RESEARCHID)
+            scrapps = scraper.get_scrapps()
+
+            keywords_and_description = scrapps[0]
+            titles = scrapps[1:]
 
-                keywords_and_description = scrapps[0]
-                titles = scrapps[1:]
+            doc = Document()
+            doc.faculty_id = faculty.faculty_id
+            doc.source = "ResearchId"
+            try:
+                doc.text = keywords_and_description.meta_data["description"]
+            except:
+                print("No description")
+                doc.text = ""
+            try:
+                doc.user_keywords = keywords_and_description.meta_data["keywords"]
+            except:
+                print("No keywords")
```

Comment:
Should this also get a default value?

Reply:
meh, if its empty then no keys words, who cares?

Reply:
What happens if you try and access it after and it doesn't have any. Is it just an empty string? I just don't want a surprise exception later on.

Reply:
yeah, ill add more